### PR TITLE
CVarify meteor behavior

### PR DESCRIPTION
**Pull request by TsjipTsjip**
## About the PR

Moves configuration values to the place they are supposed to be, I believe they were erroneously put into a component datafield in https://github.com/space-wizards/space-station-14/pull/28974 .

**Changelog**

No CL.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
All other gamerules specify all the configuration data in the rule component itself. Is there a reason it has to be a CVar?

**Comment by TsjipTsjip** [comment]
In that case it looks like I have more work to do. To my understanding the revamped meteors system is special, in the sense that it is a scheduler all by itself. That's also the only thing this PR touches as far as I'm aware.

The importance of making these things into CVars is to ensure that players who decide to host their own server can tweak the functionality provided to their preference and the needs of their playerbase, without having to hard fork from this repository and thus be required to set up a full build pipeline etc.

For my reference, which gamerules did you base this original decision on? I'll see if there's anything that can be moved to a more easily editable place there too.

**Comment by EmoGarbage404** [comment]
> The importance of making these things into CVars is to ensure that players who decide to host their own server can tweak the functionality provided to their preference and the needs of their playerbase, without having to hard fork from this repository and thus be required to set up a full build pipeline etc.
>
> For my reference, which gamerules did you base this original decision on? I'll see if there's anything that can be moved to a more easily editable place there too.

i mean basically every other gamerule uses regular component fields. i suppose survival's ramping system uses CVars so its not a huge issue however.

**Comment by deltanedas** [comment]
meteors isnt a gamerule its its own scheduler

File: Content.Server/StationEvents/MeteorSchedulerSystem.cs [hunk 1]
```diff
@@ -28,7 +34,11 @@ protected override void ActiveTick(EntityUid uid, MeteorSchedulerComponent compo
         if (Timing.CurTime < component.NextSwarmTime)
             return;
         RunSwarm((uid, component));
-        component.NextSwarmTime += RobustRandom.Next(component.MinSwarmDelay, component.MaxSwarmDelay);
+
+        var minDelay = TimeSpan.FromMinutes(_cfg.GetCVar(CCVars.MeteorSwarmMinTime));
+        var maxDelay = TimeSpan.FromMinutes(_cfg.GetCVar(CCVars.MeteorSwarmMaxTime));
```
[the remark this answers is not in the document]
Comment (TsjipTsjip) [reply Content.Server/StationEvents/MeteorSchedulerSystem.cs]: Resolved in d101825b48 .

**Comment by TsjipTsjip** [comment]
And this is what I get for not running the code. Hang on, gonna try to fix this smoothly.

File: Content.Server/StationEvents/MeteorSchedulerSystem.cs [hunk 2]
```diff
@@ -13,12 +15,22 @@ namespace Content.Server.StationEvents;
...
     protected override void Started(EntityUid uid, MeteorSchedulerComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
         base.Started(uid, component, gameRule, args);
 
-        component.NextSwarmTime = Timing.CurTime + RobustRandom.Next(component.MinSwarmDelay, component.MaxSwarmDelay);
+        _meteorMinDelay = TimeSpan.FromMinutes(_cfg.GetCVar(CCVars.MeteorSwarmMinTime));
+        _meteorMaxDelay = TimeSpan.FromMinutes(_cfg.GetCVar(CCVars.MeteorSwarmMaxTime));
+
+        _cfg.OnValueChanged(CCVars.MeteorSwarmMinTime, f => { _meteorMinDelay = TimeSpan.FromMinutes(f); });
+        _cfg.OnValueChanged(CCVars.MeteorSwarmMaxTime, f => { _meteorMaxDelay = TimeSpan.FromMinutes(f); });
```
Comment (Tayrtahn) [inline Content.Server/StationEvents/MeteorSchedulerSystem.cs:31]: OnValueChanged has an invokeImmediately option so you don't have to double up like this.

Comment (TsjipTsjip) [reply Content.Server/StationEvents/MeteorSchedulerSystem.cs]: And done. Thanks for helping me make a better system!

File: Content.Server/StationEvents/MeteorSchedulerSystem.cs [hunk 3]
```diff
@@ -13,12 +15,19 @@ namespace Content.Server.StationEvents;
...
+    private TimeSpan _meteorMaxDelay;
 
     protected override void Started(EntityUid uid, MeteorSchedulerComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
         base.Started(uid, component, gameRule, args);
 
-        component.NextSwarmTime = Timing.CurTime + RobustRandom.Next(component.MinSwarmDelay, component.MaxSwarmDelay);
+        _cfg.OnValueChanged(CCVars.MeteorSwarmMinTime, f => { _meteorMinDelay = TimeSpan.FromMinutes(f); }, true);
+        _cfg.OnValueChanged(CCVars.MeteorSwarmMaxTime, f => { _meteorMaxDelay = TimeSpan.FromMinutes(f); }, true);
+
```
Comment (EmoGarbage404) [inline Content.Server/StationEvents/MeteorSchedulerSystem.cs:29]: this needs to be in the Initialize declaration or else you're going to create duplicated subscriptions.

**Pull request merged**